### PR TITLE
`Fetch.Response._type` should access the `type` property

### PR DESCRIPTION
**Pull request by anmonteiro**
as per https://developer.mozilla.org/en-US/docs/Web/API/Response/type

### CODE REVIEW

**Comment by agarwal** [comment]
Would `type_` be a better name for this over `_type`? Variables beginning with an underscore have special meaning by the OCaml compiler. They are allowed without triggering a warning if the variable is unused. That doesn't affect this PR directly, but might be better to avoid clashing with that convention.

**Comment by glennsl** [comment]
Would type_ be a better name for this over _type?

Yeah, I think it would. At the time this was written the convention was to use leading underscore because BuckleScript used (and in some cases still uses) that to be able to reference reserved keywords in JS interop. That convention has since changed, for exactly the reason you explain and because there are better ways of dealing with this now. However, even after fixing this here is unfortunately still one case where it needs to be used, as fixing that would require a major overhaul of the API (which wouldn't be a bad idea in itself, but I certainly don't have time to do it).

Anyway, I'll fix this one here and now while we're at it. Thanks for the heads up!

File: src/Fetch.ml [hunk 1]
```diff
@@ -373,7 +373,7 @@ module Response = struct
   external redirected : t -> bool = "redirected" [@@bs.get]
   external status : t -> int = "status" [@@bs.get]
   external statusText : t -> string = "statusText" [@@bs.get]
-  external _type : t -> string = "_type" [@@bs.get]
+  external _type : t -> string = "type" [@@bs.get]
```
Comment (glennsl) [inline src/Fetch.ml]:
```suggestion
  external type_ : t -> string = "type" [@@bs.get]
```

**Review by glennsl — APPROVED** [review]
Thanks for the fix @anmonteiro !

**Pull request merged**